src/storage/src/source/s3.rs

```diff
@@ -73,12 +73,6 @@ use super::metrics::SourceBaseMetrics;
 mod metrics;
 mod notifications;
 
-struct InternalMessage {
```

Comment:
I didn't spend the time to make the S3 actually produce a nicely chunked `ByteStream`. It will currently buffer each file in memory and emit a degenerate `ByteStream` which will yield only one buffer containing the entirety of the file.

We should revisit when S3 gets productionized, the source code in here need a lot of work.

Reply:
this seems like the right choice! Can you document this in a module doc comment?